File: app/src/main/java/me/imzack/app/ender/injector/module/AllTypesPresenterModule.java
package me.imzack.app.ender.injector.module;

import me.imzack.app.ender.view.contract.AllTypesViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class AllTypesPresenterModule {

    private final AllTypesViewContract mAllTypesViewContract;

    public AllTypesPresenterModule(AllTypesViewContract allTypesViewContract) {
        mAllTypesViewContract = allTypesViewContract;
    }

    @Provides
    AllTypesViewContract provideAllTypesViewContract() {
        return mAllTypesViewContract;
    }
}

File: app/src/main/java/net/zackzhang/app/end/injector/module/AllTypesPresenterModule.kt
package net.zackzhang.app.end.injector.module

import dagger.Module
import dagger.Provides
import net.zackzhang.app.end.view.contract.AllTypesViewContract

@Module
class AllTypesPresenterModule(private val mAllTypesViewContract: AllTypesViewContract) {

    @Provides
    fun provideAllTypesViewContract() = mAllTypesViewContract
}
